### ISSUE TITLE
Signed reference expiration time is measured in seconds since epoch

### ISSUE THREAD

**Issue by jboynes** [GoogleCloudPlatform/gcloud-java#107]
Java measures time in milliseconds from the epoch not seconds and using this as a base would be simpler. The example:
```java
      Calendar cal = Calendar.getInstance();
      cal.add(Calendar.DATE, 1);
      long expiration = cal.getTimeInMillis() / 1000;
      storage.signUrl(blobInfo, expiration));
```
would become:
```java
      Calendar cal = Calendar.getInstance();
      cal.add(Calendar.DATE, 1);
      storage.signUrl(blobInfo, cal.getTime()));
```

However, given this is actually just performing an interval calculation we could also support:
```java
    storage.signUrl(blobInfo, 1, TimeUnit.DAYS);
```

Ideally we would also support the time classes added in Java 8:
```java
    storage.signUrl(blobInfo, Instant.now().plus(24, ChronoUnit.HOURS));
    storage.signUrl(blobInfo, Duration.ofDays(1));
```

**Comment by jgeewax** [comment]
Bit +1 to the `Duration.ofDays(1)` syntax :)

**Comment by aozarov** [comment]
It is in second because this is the granularity that the underlying service supports and I was hesitant to give the impression that we support something finer (even though I completely agree that for Java millis is more natural).

Indeed Java 8 gives us much better options but do we want to change it until then?

I considered `TimeUnit` but found it a bit strange to use because of its association to util.concurrent.

Another option could be to use/pass `java.util.Date`.

**Comment by eamonnmcmanus** [comment]
If we are happy to target only Java 8 then Duration is perfect. If we want to support pre-Java 8, including Android, then I think TimeUnit is the way to go. Though it happens to be in java.util.concurrent for historical reasons, that doesn't stop other APIs like java.nio and even javax.swing from using it. Leaving a raw number seems problematic; I'm sure we've all seen bugs, sometimes long-standing bugs, due to people accidentally using the wrong units.

**Comment by mziccard** [comment]
I am in favour of using `TimeUnit` and support:
```java
storage.signUrl(blobInfo, 1, TimeUnit.DAYS);
```
Should we drop the support for `storage.signUrl(blobInfo, expiration)`? There still might be some use cases for that even though I think we should just drop it.

**Comment by aozarov** [comment]
Go ahead and send a PR for it (also javadoc will not get get updated, though code snippets becomes
even less useful in my mind).

Yes, I think we should add it instead rather than keeping them both.

**Comment by aozarov** [comment]
Fixed by #243
